### PR TITLE
Add dispatch support

### PR DESCRIPTION
**Pull request by matiskay**
## How it works?

Adds a variable `DispatchSnsArn` which will be used to route the data to dispatch.  

## How to use it in a patrol rule?

Only add validation to check if `DispatchSnsArn` is set and format the message for dispatch must use [Dispatch message spec](https://github.com/mapbox/dispatch/blob/master/MESSAGE-SPEC.md).

```
module.exports.fn = (event, context, callback) => {
  if (process.env.DispatchSnsArn) {
    return githubMadePublicEvent.madePublic(event, dispatchNotify, callback);
  }

  return githubMadePublicEvent.madePublic(event, patrolNotify, callback);
};
```

### CODE REVIEW

**Comment by matiskay** [comment]
This is more than adding dispatch support now.

* Drop node 4.3.2 support.
* Transition to ES6.
* Split the functions into files
* Add `LambdaCfnDispatchRole` that allows that allows to use `DispatchSnsArn`.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file. For change
 ## Unreleased
 - updated contrib docs to include instructions to update tests for new releases
 
+## 2.0.2 2018-01-24
+- Drop node 4.3.2 support.
```
Comment (zmully) [inline CHANGELOG.md]: This will break anyone running lambda-cfn v2.x on node 4.3.2 right? So we should bump this to 3.x to prevent that?

Comment (matiskay) [reply CHANGELOG.md]: @zmully, not it will not affect any running lambda function. Because every lambda runs in its own stack.

Comment (matiskay) [reply CHANGELOG.md]: @zmully, You are right. It's better a bump a 3.x version. I will make that modification.

File: README.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ Quickly create, deploy, and manage AWS Lambda functions via AWS CloudFormation.
 
 ### Node.js
 
-Lambda-cfn (or "Lambda CloudFormation") is a Node.js project that runs on Amazon Web Service's [supported Node runtimes](https://docs.aws.amazon.com/lambda/latest/dg/current-supported-versions.html), currently Node.js v4.3.2 and v6.10.3.
+Lambda-cfn (or "Lambda CloudFormation") is a Node.js project that runs on Amazon Web Service's [supported Node runtimes](https://docs.aws.amazon.com/lambda/latest/dg/current-supported-versions.html), currently Node.js only v6.10.3.
```
Comment (zmully) [inline README.md]: This is awkwardly phrased (probably my own fault). 

> Lambda-cfn (or "Lambda CloudFormation") is a Node.js project that runs on Amazon Web Service's [supported Node runtimes](https://docs.aws.amazon.com/lambda/latest/dg/current-supported-versions.html), currently Node.js only v6.10.3.


Something like?
> Lambda-cfn (or "Lambda CloudFormation") is a Node.js project that runs on [AWS Lambda](https://docs.aws.amazon.com/lambda/latest/dg/welcome.html). Only AWS Lambda [Node runtime](https://docs.aws.amazon.com/lambda/latest/dg/current-supported-versions.html) v6.10.3 is currently supported.

File: README.md [hunk 3]
```diff
@@ -218,6 +218,38 @@ lambda-cfn save dev
 
 We're happy you want to contribute! Check out [CONTRIBUTING.MD](CONTRIBUTING.MD) for more information.
 
+## Dispatch Integration
+
+You have to set `DispatchSnsArn` variable when deploying your patrol rule and create a message following
+[dispatch message spec](https://github.com/mapbox/dispatch/blob/master/MESSAGE-SPEC.md).
+
+If you want to support both simple patrol rules and dispatch rules you check if `DispatchSnsArn` env variable is set.
```
Comment (zmully) [inline README.md]: > If you want to support both simple patrol rules and dispatch rules you check if `DispatchSnsArn` env variable is set.

This is really difficult to convey, I'm not sure if this any better, but I'm not sure if anyone knows the difference between `simple patrol rule` and `dispatch rule`.

> Dispatch support can be determined at runtime by checking if the `DispatchSnsArn` environment variable is set. For example:

File: README.md [hunk 4]
```diff
@@ -218,6 +218,38 @@ lambda-cfn save dev
 
 We're happy you want to contribute! Check out [CONTRIBUTING.MD](CONTRIBUTING.MD) for more information.
 
+## Dispatch Integration
+
```
Comment (zmully) [inline README.md]: I think we should add some more context here for people not familiar with Dispatch.

> `lambda-cfn` now supports the message router [Dispatch](https://github.com/mapbox/dispatch) which provides rich integration with Slack and GitHub. If the _optional_ `DispatchSnsArn` parameter is specified, `lambda-cfn` will grant the function permission to publish to the specified SNS Topic Arn. 

There should be something about how to use `lib/message.js` in here as well I think.

File: README.md [hunk 5]
```diff
@@ -218,6 +218,38 @@ lambda-cfn save dev
 
 We're happy you want to contribute! Check out [CONTRIBUTING.MD](CONTRIBUTING.MD) for more information.
 
+## Dispatch Integration
```
Comment (k-mahoney) [inline README.md]: For readability, I'd put the section of dispatch use and the util functions above the contribution instructions.

File: README.md [hunk 6]
```diff
@@ -218,6 +218,38 @@ lambda-cfn save dev
...
+
+You have to set `DispatchSnsArn` variable when deploying your patrol rule and create a message following
```
Comment (k-mahoney) [inline README.md]: "In order to use [`dispatch`](https://github.com/mapbox/dispatch), you have to set..."

**Pull request merged**